salt/utils/data.py

```diff
@@ -993,3 +993,40 @@ def json_query(data, expr):
         log.error(err)
         raise RuntimeError(err)
     return jmespath.search(expr, data)
+
+
+def filter_falsey(data, recurse_depth=None, ignore_types=None):
+    '''
+    Helper function to remove items from a dict or list with falsey value.
+    Removes ``None``, ``{}`` and ``[]``, 0, '' (but does not remove ``False``).
+    Recurses into subdicts or sublists if ``recurse`` is set to ``True``.
+
+    :param dict/list data: Source dict or list to process.
+    :param int recurse_depth: Recurse this many levels into values that are dicts
+        or lists to also process those. Default: 0 (do not recurse)
+    :param list ignore_types: Contains types that can be falsey but must not
+        be filtered. Default: Only booleans are not filtered.
+
+    :return type(data)
+    '''
```

Comment:
We need to add a `.. version-added:: Neon` since this will be new to the Neon release.

Reply:
You are right, I totally forgot about that. Added.